src/lib/strings.h

```diff
@@ -23,6 +23,34 @@ void free_string(char **s);
 /* Join strings from string list separated by the separator */
 char *string_join(const char *separator, struct list *string);
 
+/*
+ * strtoi_err: Safely convert and string to integer avoiding overflows
+ *
+ * The strtol function is commonly used to convert a string to a number and
+ * the result is frequently stored in an int type, but type casting a long to
+ * an int can cause overflows.
+ *
+ * This function returns negative error codes based on the errno table:
+ * -ERANGE is returned when the string is out of range for int value
+ * -EINVAL is returned when the string isn't a valid number or has any invalid
+ * trailing character.
+ */
+int strtoi_err(const char *str, int *value);
+
+/*
+ * strtoi_err: Safely convert and string to integer avoiding overflows.
```

Comment:
wrong function name in the comment strtoi_err -> strtoi_err_endptr (and same typo as above)